src/vm_state/helpers.rs

```diff
@@ -205,6 +205,18 @@ impl<
             .decommittment_processor
             .prepare_to_decommit(monotonic_cycle_counter, partial_query)?;
 
+        if query.is_fresh {
```

Comment:
let's add a comment, that "we should never execute decommit opcode twice in the same memory page"